format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -46,7 +48,13 @@
                                                                                     @Nullable Schema schema) {
 
     RecordReader<LongWritable, Text> delegate = (new TextInputFormat()).createRecordReader(split, context);
-    String delimiter = context.getConfiguration().get(DELIMITER);
+    String delimeter = context.getConfiguration().get(DELIMITER);
+    String regex = delimeter;
+    boolean enableQuotesValue = context.getConfiguration().getBoolean(ENABLE_QUOTES_VALUE, false);
+    if (enableQuotesValue) {
+      regex = String.format("%s(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)", regex);
```

Comment:
I don't really understand this regex and can't tell that it behaves correctly, can you add some comments about it? We also need more unit tests, like testing what happens with multiple quoted values on the same line, odd number of quotes, etc.